lib/translators/static.js

```diff
@@ -17,13 +17,15 @@ const parseYaml = require("js-yaml").safeLoad;
  * @param {Array} [translatorOptions] Configuration options
  * @returns {Promise} Promise resolving with a dependency tree
  */
-function generateDependencyTree(dirPath, translatorOptions) {
+function generateDependencyTree(dirPath, translatorOptions = []) {
 	const depFilePath = translatorOptions[0] || path.join(dirPath, "projectDependencies.yaml");
 
 	return new Promise(function(resolve, reject) {
 		fs.readFile(depFilePath, function(err, buffer) {
 			if (err) {
-				reject(err);
+				reject(new Error(
+					`[static translator] Failed to locate projectDependencies.json for "${dirPath}"`)
```

Comment:
There are all kinds of low-level file system errors that might happen here. Therefore the actual fs-error should be part of the new error message.

Example:
https://github.com/SAP/ui5-project/blob/97e96f3677050901ba097b3fadaea07e95914019/lib/translators/npm.js#L231-L232

Reply:
I think it should read
> `Failed to locate projectDependencies.yaml at path "${dirPath}"`

rather than
> `Failed to locate projectDependencies.json for "${dirPath}"`

Reply:
Nice, thanks for great feedback. Should be fine now.